media-gfx/hydrus/hydrus-441.ebuild

```diff
@@ -0,0 +1,154 @@
+		dev-python/requests[${PYTHON_MULTI_USEDEP}]
+		dev-python/send2trash[${PYTHON_MULTI_USEDEP}]
+		dev-python/service_identity[${PYTHON_MULTI_USEDEP}]
+		dev-python/six[${PYTHON_MULTI_USEDEP}]
+		dev-python/twisted[${PYTHON_MULTI_USEDEP}]
+		media-libs/opencv[python,${PYTHON_MULTI_USEDEP}]
+
+		>=dev-python/QtPy-1.9.0-r4[pyside2,${PYTHON_MULTI_USEDEP}]
+		dev-python/beautifulsoup:4[${PYTHON_MULTI_USEDEP}]
+
+		cloudscraper? ( dev-python/cloudscraper[${PYTHON_MULTI_USEDEP}] )
+		ffmpeg? ( media-video/ffmpeg )
+		lz4? ( dev-python/lz4[${PYTHON_MULTI_USEDEP}] )
+		miniupnpc? ( net-libs/miniupnpc )
+		mpv? (
+			 media-video/mpv[libmpv,${PYTHON_MULTI_USEDEP}]
+			 dev-python/python-mpv[${PYTHON_MULTI_USEDEP}]
+		)
+		socks? (
+				|| ( dev-python/requests[socks5,${PYTHON_MULTI_USEDEP}]
+					dev-python/PySocks[${PYTHON_MULTI_USEDEP}] )
+		)
+	')
+"
+BDEPEND="
+	${RDEPEND}
+	$(python_gen_cond_dep '
+		test? (
+			dev-python/httmock[${PYTHON_MULTI_USEDEP}]
+			dev-python/mock[${PYTHON_MULTI_USEDEP}]
+			dev-python/nose[${PYTHON_MULTI_USEDEP}]
+			dev-python/unittest2[${PYTHON_MULTI_USEDEP}]
+		)
+	')
+"
+
+# Delete files only needed for testing
+delete_test_files() {
+	rm test.py
+	rm hydrus/hydrus_test.py
+	rm -r hydrus/test/
+	rm -r static/testing/
+}
+
+PATCHES=(
+	"${FILESDIR}/userpath-in-local-share.patch"
+)
+
+src_prepare() {
+	default
+
+	# If tests will run, leave the files until tests are run.
+	# They will be deleted before installing the package.
+	if ! use test; then
+		delete_test_files
+	fi
+
+	# Contains pre-built binaries for other systems and a broken swf renderer for linux
+	rm -r bin/
+	# Build files used for CI, not actually needed
+	rm -r static/build_files
+	# Duplicate license file, not needed
+	rm license.txt
+	# Python requirements files, not needed
+	rm requirements_*.txt
+}
+
+src_compile() {
+	python_optimize "${S}"
+}
+
+src_test() {
+	# The tests use unittest, but are run with a custom runner script.
+	# QT_QPA_PLATFORM is required to make them run without X
+	export QT_QPA_PLATFORM=offscreen
+	"${EPYTHON}" "${S}/test.py" || die "Tests failed"
+}
+
+src_install() {
+	DOC="/usr/share/doc/${PF}"
+	elog "Hydrus includes an excellent manual, that can either be viewed at"
+	elog "${DOC}/html/help/index.html"
+	elog "or accessed through the hydrus help menu."
+
+	mv "help my client will not boot.txt" "help_my_client_will_not_boot.txt"
+
+	DOCS="COPYING README.md Readme.txt help_my_client_will_not_boot.txt db/"
+	HTML_DOCS="${S}/help/"
+	einstalldocs
+
+	if use test; then
+		# Delete files only needed for tests now. No need to install them.
+		# If the tests didn't run, the files have been deleted already.
+		delete_test_files
+	fi
+
+	# These files are copied into DOC
+	rm COPYING README.md Readme.txt help_my_client_will_not_boot.txt
+	rm -r help/ db/
+	# The program expects to find documentation here, so add a symlink to DOC
+	ln -s "${DOC}/html/help" help
+
+	insopts -m0755
+	insinto /opt/${PN}
```

Comment:
I don't know if I'd bother with ${PN} here, it's not like we're going to copy it to another ebuild